base/uk.ac.stfc.isis.ibex.ui.logplotter/src/uk/ac/stfc/isis/ibex/ui/logplotter/Perspective.java

```diff
@@ -51,7 +57,48 @@ public Image image() {
 	public void createInitialLayout(IPageLayout layout) {
 		super.createInitialLayout(layout);
 		
-		layout.setEditorAreaVisible(true);
-		lockView(layout, "org.csstudio.trends.databrowser.waveformview.WaveformView");		
+		lockView(layout, "org.csstudio.trends.databrowser.waveformview.WaveformView");
+		
+		layout.addStandaloneView(EmptyLogPlotterView.ID, false, IPageLayout.RIGHT, 0.1f, "uk.ac.stfc.isis.ibex.ui.perspectives.PerspectiveSwitcher");
+		
+		
+		//This part listener will be called to open the empty view when all graphs are closed
+		PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage().addPartListener(new IPartListener() {
+			
+			@Override
+			public void partOpened(IWorkbenchPart part) {
+			}
+			
+			@Override
+			public void partDeactivated(IWorkbenchPart part) {
+			}
+			
+			@Override
+			public void partClosed(IWorkbenchPart part) {
+				checkEditorEmpty();
+			}
+			
+			@Override
+			public void partBroughtToTop(IWorkbenchPart part) {
+			}
+			
+			@Override
+			public void partActivated(IWorkbenchPart part) {
+			}
+			
+			private void checkEditorEmpty() {
+				IWorkbenchPage activePage = PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage();
+				if (activePage != null) {
+					if (activePage.getEditorReferences().length == 0 ) {
+						activePage.setEditorAreaVisible(false);
+						try {
+							activePage.showView(EmptyLogPlotterView.ID);
+						} catch (PartInitException e) {
+							e.printStackTrace();
+						}
```

Comment:
Could you maybe do a small refactor here, so that there aren't so many nested brackets, something like:
```
if (activePage == null || activePage.getEditorReferences().length > 0) {
return;
}

activePage.setEditorAreaVisible(false);
...
```